### PR TITLE
Issue #2271: Start domain request bug (multiple can get created) - [HOTGOV]

### LINKED ISSUE #2271: "Start a new domain request" results in creating two domain requests
### Current Behavior

When user do not have any new domain requests in your domain request table. Two domain requests are created when user "Start a new domain request".

### Expected Behavior

"Start a new domain request" results in creating ONLY one request regardless of how many domain requests the user has.

### Steps to Reproduce

1. User does not have any new domains requests.
2. User selects "Start a new domain request"
3. Two domain requests are created. View domain request table.

### Environment

Local and Staging

### Additional Context

See[ slack thread](https://cisa-corp.slack.com/archives/C05BGB4L5NF/p1717559447810209)

Troubleshooting:
- The "You're about to start your .gov request" disclaimer screen seems to trigger the creation of a request in this case.

### Issue Links

_No response_

### PR DESCRIPTION
**Pull request by rachidatecs**
## Ticket

Resolves #2271 

## Changes

- Creating a new request when the sesh is empty causes the domain_request getter (called from GET) to create a request, while the POST also creates a request = 2 for 1: This was fixed by revising the GET
- Using the back button and hitting Continue would create requests ad nauseam (Into page 'continue' -> back button -> intro page 'continue' = 2 for 1). This was fixed by ~timing the interval between POSTs from that 'continue' button; if they're less that 5 secs, it indicates either shenanigans or back button use (in my opinion)~ adding a special global session check.

## Context for reviewers

One known issue: Click Create, Continue past intro, fill in org type, save, back button from step 2, back button again, click Continue past intro: an extra new request is created. This can be solved by removing `else:  self.request.session["new_request"] = True` in goto in the wizard, but I feel this could make it all a bit flimsier (more risk of an edge case in my blind spot where a user overwrites an existing request, which would be critical). cc @abroddrick 

## Setup

- Click create request when you have no other new requests. Click continue. Click back to the homepage. 1 new request should have been created (as opposed to 2)
- Click create request, click continue, click back button, click continue, do that a few times. 1 new request should have been created as opposed to multiple.
- Make sure session works as expected otherwise (edit directs to the right requests, etc.)
- Click create new request again, a new request should have been created.

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [x] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [x] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [x] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Review by abroddrick — CHANGES_REQUESTED** [review]
Sorry, but I think this will just result in future tech debt and other issues. We should not use time to determine if we make a new request in our session.

**Comment by rachidatecs** [comment]
> Sorry, but I think this will just result in future tech debt and other issues. We should not use time to determine if we make a new request in our session.

Revised

File: src/registrar/views/domain_request.py [hunk 1]
```diff
@@ -435,6 +442,16 @@ def get_step_list(self) -> list:
...
 
     def goto(self, step):
+        if step == "generic_org_type":
+            # We need to avoid creating a new domain request if the user
+            # clicks the back button
+            self.request.session["new_request"] = False
+        else:
+            # Reset the above logic to be extra safe;
+            # we do not want to stumble into a situation where a user
+            # unkowingly overrites when she thinks she's working on a
```
Comment (abroddrick) [inline src/registrar/views/domain_request.py:452]:
```suggestion
            # unknowingly overwrites when she thinks she's working on a
```

**Review by abroddrick — APPROVED** [review]
This solution does feel a bit hacky, but I think it does solve the initial bug that this ticket was made for. It definitely isn't perfect given there are still ways (as noted in the context for reviewers) to still get multiple applications started when the user is only intending to make one. 

I am fine with approving for now, but recommend leaving this PR open for another day or so in case someone else has an idea of a better solution. @dave-kennedy-ecs was thinking of looking at this one so if there is a better way found please feel free to ignore this approval.

**Comment by rachidatecs** [comment]
> This solution does feel a bit hacky, but I think it does solve the initial bug that this ticket was made for. It definitely isn't perfect given there are still ways (as noted in the context for reviewers) to still get multiple applications started when the user is only intending to make one.
> 
> I am fine with approving for now, but recommend leaving this PR open for another day or so in case someone else has an idea of a better solution. @dave-kennedy-ecs was thinking of looking at this one so if there is a better way found please feel free to ignore this approval.

ICYMI: else:  Removing self.request.session["new_request"] = True in goto in the wizard might cover the last scenario, but am not sure it's worth dumping that safety net to cover a far-fetched edge case

**Review by dave-kennedy-ecs — APPROVED** [review]
Reviewed code with Rachid and independently.  Tested locally.  Nice work!

**Pull request merged**